string part
string name
---
bool collisionState
